### PR TITLE
defining profile in useProfileContext to get the latest profile data …

### PR DESCRIPTION
**Pull request by hclarkEnq**
Description 
Fixing a bug in the patient profile where the patients data was not being reflected immediately upon edits or changes done in the demographics card using any table or card where the information is changeable.

## Steps to verify

1. Login
1. Navigate to any patient profile after searching for a patient
2. navigate to demographics and edit to change the "name" of the patient
3. observe the changes reflected immediately in the summary card in the patient profile at the top of the page

### CODE REVIEW

File: apps/modernization-ui/src/pages/patient/profile/ProfileContext.tsx [hunk 1]
```diff
@@ -19,9 +20,9 @@ export function useProfileContext() {
 }
 
 export const ProfileProvider = ({ children, id }: any) => {
-    const { refetch } = usePatientProfile(id);
+    const { refetch, profile } = usePatientProfile(id);
 
     const changed = () => refetch();
 
-    return <ProfileContext.Provider value={{ changed }}>{children}</ProfileContext.Provider>;
+    return <ProfileContext.Provider value={{ changed, profile }}>{children}</ProfileContext.Provider>;
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/pages/patient/profile/ProfileContext.tsx:27]: issue:  combining `patient` and `summary` into one object will cause the entire page to re-render.  An alternative approach would be to have `usePatientProfile`return the `patient` and `summary` as two separate items in `PatientProfile.tsx`

`const {patient, summary} = usePatientProfile()`

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
Putting the `patient` in the ProfileContext is okay however, adding the profile will cause the entire page to re-render when the summary changes.

File: apps/modernization-ui/src/pages/patient/profile/ProfileContext.tsx (hunk 1 above)
Comment (hclarkEnq) [reply apps/modernization-ui/src/pages/patient/profile/ProfileContext.tsx:27]: got it

File: apps/modernization-ui/src/pages/patient/profile/addresses/AddressesTable.tsx [hunk 2]
```diff
@@ -129,17 +127,18 @@ export const AddressesTable = ({ patient }: Props) => {
...
+                variables: {
+                    patient: patient?.id,
+                    page: {
+                        pageNumber: currentPage - 1,
+                        pageSize: TOTAL_TABLE_DATA
+                    }
+                },
+                notifyOnNetworkStatusChange: true
+            });
+    }, [currentPage, patient]);
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/pages/patient/profile/addresses/AddressesTable.tsx:141]: question: Is `patient` needed as a dependency? It's passed in and not part of the state of this component

**Review by adamloup-enquizit — APPROVED** [review]

File: apps/modernization-ui/src/pages/patient/profile/addresses/AddressesTable.tsx (hunk 2 above)
Comment (hclarkEnq) [reply apps/modernization-ui/src/pages/patient/profile/addresses/AddressesTable.tsx:141]: Yes because patient.ID is being passed into the API

**Pull request merged**